website/docs/r/lb_backend_address_pool.html.markdown

```diff
@@ -52,6 +52,10 @@ The following arguments are supported:
   
 * `loadbalancer_id` - (Required) The ID of the Load Balancer in which to create the Backend Address Pool. Changing this forces a new resource to be created.
 
+* `synchronous_mode` - (Optional) The backend address synchronous mode for the Backend Address Pool. Possible values are `Automatic` and `Manual`. This is required with `virtual_network_id`. Changing this forces a new resource to be created.
```

Comment:
This description isn't clear - what does this field do?

Reply:
Hi @tombuildsstuff thanks for your time. The following is an introduction to `synchronous_mode` from the service team. May I ask you to give me the description suggestion for the `synchronous_mode` in TF?


What is syncMode?
```
The syncMode property is a parameter that you can specify when you create a backend pool by using IP addresses and virtual network IDs. This property must be set when using cross-subscription frontends or backends. It has two possible values: “Automatic” or “Manual”.

In addition, this property replaces the concept of NIC-based or IP-based backend pools. As a result, backend pools with the syncMode property configured are a distinct type of backend pool, separate from NIC or IP-based backend pools. Backend pools can either be exclusively NIC-based, IP-based, or syncMode enabled.
```

When should I use syncMode: “Automatic”?
```
With syncMode configured as “Automatic”, backend pool instances are synchronized with the load balancer configuration. As a result, changes to the backend pool instances are automatically reflected in the load balancer’s backend pool configuration. This is particularly relevant when leveraging virtual machine scale sets (VMSS) in the backend pool. When the VMSS scales in/out, the backend pool members will automatically be added or removed from the pool accordingly.

Like NIC (network interface cards) based backend pools, if syncMode is set to “Automatic”, then each backend instance’s NIC must also reference the load balancer backend pool. As a result, backend instances are added to “Automatic” syncMode backend pools by updating the NIC resource’s reference to the load balancer.
```

When should I use syncMode: “Manual”?
`With syncMode configured as “Manual”, backend pool instances are not synchronized with the load balancer configuration. This mode allows you to create a backend pool with pre-provisioned private IP addresses that can be used for scenarios such as disaster recovery, active-passive, or dynamic provisioning. When leveraging “Manual” syncMode backend pools, you are responsible for updating the backend pool when any changes to your backend instances occur, such as in the case of VMSS autoscaling.`

Reply:
@sinbai, I think the point that @tombuildsstuff is trying to get at here is to reflect on what this feature actually does and how it interacts with the service, although we're testing that we can add this property in an API request and receive it back in the response I don't think the test is sufficient to fully validate the functionality of this. This is particular concern for load balancers because of the way they interact with other resources.